tools/clang/tools/hlsl2spirv/hlsl2spirv.cpp

```diff
@@ -0,0 +1,216 @@
+//===- hlsl2spirv.cpp ---- Entry point for hlsl2spirv program -------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#include "dxc/Support/Global.h"
+#include "dxc/Support/Unicode.h"
+#include "dxc/Support/WinIncludes.h"
+#include "dxc/Support/HLSLOptions.h"
+#include "dxc/Support/dxcapi.use.h"
+#include "dxc/dxcapi.h"
+#include "dxc/dxcapi.internal.h"
+
+#include "llvm/Option/OptTable.h"
+#include "llvm/Support/raw_ostream.h"
+
+#include "spirv-tools/libspirv.hpp"
+
+using namespace dxc;
+using namespace llvm;
+using namespace llvm::opt;
+using namespace hlsl::options;
+
+namespace {
+void ConvertIDxcBlobToUint32(const CComPtr<IDxcBlob> &blob,
+                             std::vector<uint32_t> *binaryWords) {
+  if (!blob)
+    return;
+  size_t num32BitWords = (blob->GetBufferSize() + 3) / 4;
+  std::string binaryStr((char *)blob->GetBufferPointer(),
+                        blob->GetBufferSize());
+  binaryStr.resize(num32BitWords * 4, 0);
+  binaryWords->resize(num32BitWords, 0);
+  memcpy(binaryWords->data(), binaryStr.data(), binaryStr.size());
+}
+} // anonymous namespace
+
+class Hlsl2SpirvContext {
+private:
+  DxcDllSupport &dllSupport;
+  DxcOpts &options;
+  CComPtr<IDxcBlob> binaryBlobPtr;
+  CComPtr<IDxcBlob> spirvAssemblyBlobPtr;
+  std::string spirvAssembly;
+
+public:
+  Hlsl2SpirvContext(DxcDllSupport &dxcSupport, DxcOpts &opts)
+      : dllSupport(dxcSupport), options(opts) {}
+
+  void Compile();
+  void DisassembleResults();
```

Comment:
I'd suggest to just rename these methods to `Disassmeble()` and `Report()`. It's clear enough from the name. Adding the `Results()` doesn't provide more information IMHO.